packages/steveo/test/consumer/kafka_test.ts

```diff
@@ -358,4 +358,60 @@ describe('runner/kafka', () => {
     const expectedContext = getContext(messagePayload);
     expect(context, 'expected context').to.deep.equals(expectedContext);
   });
+
+  it('should unpack message value before forward payload to middleware', async () => {
+    const subscribeStub = sinon.stub();
+    const anotherRegistry = {
+      getTask: () => ({
+        publish: () => {},
+        subscribe: subscribeStub,
+      }),
+      emit: sandbox.stub(),
+      events: {
+        emit: sandbox.stub(),
+      },
+    };
+    const messagePayload: Record<string, string> = {
+      my: 'payload'
```

Comment:
Let's also add a test for when the payload is a string. That is the edge case that this PR is aiming to fix, yeah?

Reply:
Only for the producer. The consumer will always receive a Json parseable payload.